cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java

```diff
@@ -217,6 +220,15 @@ private void collectBranchesData(
 
         List<EndBranchDataContainer> branches = executionRuntimeServices.getFinishedChildBranchesData();
         for (EndBranchDataContainer branch : branches) {
+
+            //first we check that no exception was thrown during the execution of the branch
+            String branchException = branch.getException();
+            if (StringUtils.isNotEmpty(branchException)) {
+                logger.error("There was an error running branch: " +
+                        executionRuntimeServices.getBranchId() + " Error is: " + branchException);
+                throw new RuntimeException(BRANCH_EXCEPTION_PREFIX + ": \n" + branchException);
```

Comment:
Do we want the behavior to be that in case one of the branches fail we throw an exception and stop execution?